packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-credit-button-strategy.ts

```diff
@@ -98,6 +124,183 @@ export default class PaypalCommerceCreditButtonStrategy implements CheckoutButto
+        // on this stage we don't have access to valid customer's address accept shipping data
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+        await this._store.dispatch(this._consignmentActionCreator.updateAddress(address));
+
+        const shippingOption = this._getShippingOptionOrThrow();
+
+        await this._store.dispatch(this._consignmentActionCreator.selectShippingOption(shippingOption.id));
+        await this._updateOrder();
+    }
+
+    private async _updateOrder(): Promise<void> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const consignment = state.consignments.getConsignmentsOrThrow()[0];
+
+        await this._paypalCommerceRequestSender.updateOrder({
+            availableShippingOptions: consignment?.availableShippingOptions,
+            cartId: cart.id,
+            selectedShippingOption: consignment?.selectedShippingOption,
+        });
+    }
+
+    private _getShippingOptionOrThrow(selectedShippingOptionId?: string): ShippingOption {
+        const state = this._store.getState();
+        const consignment = state.consignments.getConsignmentsOrThrow()[0];
+
+        const availableShippingOptions = consignment?.availableShippingOptions || [];
+
+        const recommendedShippingOption = availableShippingOptions.find(option => option.isRecommended);
+        const selectedShippingOption = selectedShippingOptionId
+            ? availableShippingOptions.find(option => option.id === selectedShippingOptionId)
+            : availableShippingOptions.find(option => option.id === consignment?.selectedShippingOption?.id);
+
+        const shippingOptionToSelect = selectedShippingOption || recommendedShippingOption;
+
+        if (!shippingOptionToSelect) {
+            throw new Error('Your order can\'t be shipped to this address');
+        }
+
+        return shippingOptionToSelect;
+    }
+
+    private _transformAddress(address?: Partial<BillingAddressRequestBody>): BillingAddressRequestBody {
+        return {
+            firstName: address?.firstName || 'Fake',
+            lastName: address?.lastName || 'Fake',
+            email: address?.email || 'fake@fake.fake',
+            phone: '',
+            company: '',
+            address1: address?.address1 || 'Fake street',
+            address2: '',
+            city: address?.city || '',
+            countryCode: address?.countryCode || '',
+            postalCode: address?.postalCode || '',
+            stateOrProvince: '',
+            stateOrProvinceCode: address?.stateOrProvinceCode || '',
+            customFields: [],
+        };
+    }
+
+    private _getValidAddress(
+        payerName: PayPalOrderDetails['payer']['name'],
+        email: string,
+        address: PayPalOrderAddress,
+    ) {
+        return this._transformAddress({
```

Comment:
Is an address with all `fakes` a valid address?

Reply:
Yes it is satisfies all required fields for address